doc/luasnip.txt

```diff
@@ -1453,6 +1454,152 @@ callback will have access to the `POSTFIX_MATCH` field as well.
+    - `display`: `display(snippets:string)`
+
+
+Benefits include: syntax highlighting, searching, and customizability.
+
+Simple Example:
+
+>
+    sl.open()
+<
+
+
+<div class="figure">
+<img src="https://user-images.githubusercontent.com/43832900/204893019-3a83d6bc-9e01-4750-bdf4-f6af967af807.png" title="fig:"/>
+<p class="caption">default</p>
+</div>
+
+Customization Examples:
+
+>
+    -- making our own snip_info
+    local function snip_info(snippet)
+        return { name = snippet.name }
+    end
+    
+    -- using it
+    sl.open({snip_info = snip_info})
+<
+
+
+<div class="figure">
+<img src="https://user-images.githubusercontent.com/43832900/204893340-c7296a70-370a-4ad3-8997-23887f311b74.png" title="fig:"/>
+<p class="caption">snip_info</p>
+</div>
+
+>
+    -- making our own printer
+    local function printer(snippets)
+        local res = ""
+    
+        for ft, snips in pairs(snippets) do
+            res = res .. ft .. "\n"
+            for _, snip in pairs(snips) do
+                res = res .. "    " .. "Name: " .. snip.name .. "\n"
+                res = res .. "    " .. "Desc: " .. snip.description[1] .. "\n"
+                res = res .. "    " .. "Trigger: " .. snip.trigger .. "\n"
+                res = res .. "    ----" .. "\n"
+            end
+        end
+    
+        return res
+    end
+    
+    
+    -- using it
+    sl.open({printer = printer})
+<
+
+
+<div class="figure">
+<img src="https://user-images.githubusercontent.com/43832900/204893406-4fc397e2-6d42-43f3-b52d-59ac448e764c.png" title="fig:"/>
+<p class="caption">printer</p>
+</div>
+
+>
+    -- making our own display
+    local function display(printer_result)
+        -- horizontal split
+        vim.cmd("new")
+    
+        -- get buf and win handle
+        local buf = vim.api.nvim_get_current_buf()
+        local win = vim.api.nvim_get_current_win()
+    
+        -- setting window and buffer options
+        vim.api.nvim_win_set_option(win, "foldmethod", "indent")
+        vim.api.nvim_buf_set_option(buf, "filetype", "javascript")
+    
+        vim.api.nvim_buf_set_option(buf, "buftype", "nofile")
+        vim.api.nvim_buf_set_option(buf, "bufhidden", "wipe")
+        vim.api.nvim_buf_set_option(buf, "buflisted", false)
+    
+        vim.api.nvim_buf_set_name(buf, "Custom Display buf " .. buf)
+    
+        -- dump snippets
+        local replacement = vim.split(printer_result .. "\nCustom Display!", "\n")
+        vim.api.nvim_buf_set_lines(buf, 0, 0, false, replacement)
+    end
+    
+    -- using it
+    sl.open({display = display})
+<
+
+
+<div class="figure">
+<img src="https://user-images.githubusercontent.com/43832900/204893532-aa702617-315b-4f03-b3d1-a2256e171174.png" title="fig:"/>
+<p class="caption">display</p>
+</div>
+
+An `options` table with a `display` option (function) is also available in
+order to customize how you want the buffer/window/tab to look like. We’ll
+take a deeper dive in the example below.
```

Comment:
Much better!!

I think this section is a bit unclear though, maybe first introduce the `options`-table (+motivation), and then, in a list, mention the different functions in that table. (maybe also motivate why `display` is included from the get-go, should be easy with the big block of code before :D) (ouh, maybe construct the previous custom display in such a way that it can also be achieved via arguments to `options.display`, then its purpose should be immediately clear)

Reply:
Is this https://github.com/L3MON4D3/LuaSnip/pull/652/commits/82680e6306e16e63cba3af4225da2a3a5537e9a5 more what you had in mind?

Reply:
Yeah, I like it👍
I think as-is, the text for that bullet point is not in the list, some indent should fix that (might be wrong though, on mobile atm)